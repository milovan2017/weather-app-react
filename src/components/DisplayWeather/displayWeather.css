.weatherCont,
.card {
  background: transparent;
  border: none;
}
