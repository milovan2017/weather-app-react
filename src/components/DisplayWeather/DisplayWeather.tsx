import React from "react";
import { Row, Col, Card } from "react-bootstrap";
import { DayCard } from "../DayCard/DayCard";
import { WeatherData } from "../../types";
import "./displayWeather.css";

interface IDisplayWeatherProps {
  weatherData?: WeatherData;
}

const DisplayWeather: React.FC<IDisplayWeatherProps> = ({ weatherData }) => {
  return (
    <>
      <div className="weatherCont">
        {weatherData ? (
          <>
            <Col
              md={{ span: 2, offset: 5 }}
              mt={2}
              className="headerCont text-center"
            >
              <Card>
                <Card.Body>
                  <Card.Title className="text-muted">
                    {weatherData?.header.headerDate}
                  </Card.Title>
                  <h1 className="text-light">{weatherData.header.avgTemp}Â°</h1>
                </Card.Body>
              </Card>
            </Col>
            <Col md={{ span: 6, offset: 3 }} mt={2}>
              <Row>
                {weatherData?.datesList.map((dayInfo, index) => {
                  return (
                    <React.Fragment key={index}>
                      <DayCard dayInfo={dayInfo} />
                    </React.Fragment>
                  );
                })}
              </Row>
            </Col>
          </>
        ) : (
          <Col className="text-center">No Data</Col>
        )}
      </div>
    </>
  );
};

export default DisplayWeather;
