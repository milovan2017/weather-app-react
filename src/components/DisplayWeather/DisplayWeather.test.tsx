import DisplayWeather from "./DisplayWeather";
import { render, cleanup } from "@testing-library/react";

afterEach(cleanup);

const mockDisplayWeatherData = {
  header: {
    headerDate: "Mar 31 - Apr 6 2022",
    avgTemp: 38,
  },
  datesList: [
    {
      day: "",
      averageTemp: 0,
    },
  ],
};

describe("DisplayWeather.tsx", () => {
  it("renders DisplayWeather without crashing", () => {
    const { container } = render(
      <DisplayWeather weatherData={mockDisplayWeatherData} />
    );
    expect(container).toMatchSnapshot();
  });

  it("contains date info in the header", () => {
    const { container } = render(
      <DisplayWeather weatherData={mockDisplayWeatherData} />
    );
    expect(container.textContent).toContain(
      mockDisplayWeatherData.header.headerDate
    );
  });

  it("contains average temperature info", () => {
    const { container } = render(
      <DisplayWeather weatherData={mockDisplayWeatherData} />
    );
    expect(container.textContent).toContain(
      mockDisplayWeatherData.header.avgTemp.toString()
    );
  });
});
