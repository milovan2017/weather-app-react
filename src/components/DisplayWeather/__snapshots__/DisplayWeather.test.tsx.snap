// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayWeather.tsx renders DisplayWeather without crashing 1`] = `
<div>
  <div
    class="weatherCont"
  >
    <div
      class="headerCont text-center col-md-2 offset-md-5"
      mt="2"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <div
            class="text-muted card-title h5"
          >
            Mar 31 - Apr 6 2022
          </div>
          <h1
            class="text-light"
          >
            38
            °
          </h1>
        </div>
      </div>
    </div>
    <div
      class="col-md-6 offset-md-3"
      mt="2"
    >
      <div
        class="row"
      >
        <div
          class="p-0 text-center col"
        >
          <span
            class="text-muted"
          />
          <h2
            class="text-light"
          >
            0
            °
          </h2>
        </div>
      </div>
    </div>
  </div>
</div>
`;
